yabt/buildcontext.py

```diff
@@ -384,26 +384,26 @@ def build_graph(self, run_tests: bool=False):
 
         def build(target: Target):
             """Build `target` if it wasn't built already, and mark it built."""
+            # avoid rebuilding built target
             if target.name in built_targets:
                 target.done()
-            else:
-                try:
-                    self.build_target(target)
-                except Exception as ex:
-                    target.fail()
-                    fatal('`{}\': {}', target.name, ex)
+                return
+            try:
+                self.build_target(target)
```

Comment:
- how does this build target / test_target thing work
- what if you want to run test for all targets, including the built ones (say because the hardware you are running on is different)